driver-core/src/main/com/mongodb/client/model/search/SearchOperator.java

```diff
@@ -30,6 +35,17 @@
  */
 @Evolving
 public interface SearchOperator extends Bson {
+    /**
+     * Returns a base for a {@link SearchOperator} that may combine multiple {@link SearchOperator}s.
+     * Combining {@link SearchOperator}s affects calculation of the relevance score.
+     *
+     * @return A base for a {@link SearchOperator}.
+     * @mongodb.atlas.manual atlas-search/compound/ compound operator
+     */
+    static CompoundSearchOperatorBase compound() {
```

Comment:
Is an empty compound a legal search, or is at least one clause required?  Probably not a big deal either way, but maybe document it if it's not legal.

Reply:
It's not legal, and this is the answer to the [previous question](https://github.com/mongodb/mongo-java-driver/pull/928#discussion_r861356847) about `CompoundSearchOperatorBase`. We need that interface specifically because `SearchOperator.compound()` must not return a `SearchOperator` because such operator would not have been a valid one.

Reply:
> The Should part suggests that it represents only the should clause of the compound operator

`Mustable`, `MustNotable`, `Shoudable`, `Filterable` ? :trollface:

Reply:
>  I will point out examples of where I think that could be done in separate comments.

I added those docs in https://github.com/mongodb/mongo-java-driver/pull/928/commits/8aeed1fc9fe35bd5f68c775c123d7375b3b4cb75.